lib/init-steps/s1-analyze-project.js

```diff
@@ -36,24 +36,19 @@ const projectStage = [
     'valid'
 ]
 
-function run(mobileProjectID){
-    let projectPath = process.cwd()
+function run(initInfo){
     return new Promise((resolve, reject)=>{
-        if(fs.existsSync(projectPath)){
-            let initInfo = {
-                projectPath: projectPath,
-                mobileProjectID: mobileProjectID,
-                backupAWSMobileJSDirPath: undefined,
-                projectInfo: getProjectInfo(projectPath),
-                projectConfig: getProjectConfig(projectPath),
-                backendProject: getBackendProject(projectPath),
-                packageJson: getPackageJson(projectPath)
-            }
+        if(fs.existsSync(initInfo.projectPath)){
+            initInfo.backupAWSMobileJSDirPath = undefined
+            initInfo.projectInfo = getProjectInfo(initInfo.projectPath)
```

Comment:
What happen is getProjectInfo fails?

Reply:
getProjectInfo will examine the project-info.json file, if it does not exist, or if the Json parser fails, or the init records is not present in the file, the getProjectInfo function will return undefined. so the initInfo.projectInfo = undefined. that means, the current project that is being init'ed does not contain valid project info, and the following steps will acting accordingly.